x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/rest/action/RestPreviewTransformAction.java

```diff
@@ -24,7 +25,9 @@
 
     @Override
     public List<Route> routes() {
-        return List.of(new Route(POST, TransformField.REST_BASE_PATH_TRANSFORMS + "_preview"));
+        return List.of(
+            new Route(POST, TransformField.REST_BASE_PATH_TRANSFORMS + "_preview"),
+            new Route(POST, TransformField.REST_BASE_PATH_TRANSFORMS_BY_ID + "_preview"));
```

Comment:
I think `_transform/id/_preview` is conceptually a _GET_ request.

Looking at data feeds I see both (_POST_, _GET_) [specified](https://www.elastic.co/guide/en/elasticsearch/reference/current/ml-preview-datafeed.html) for preview (user friendliness?). We should be consistent.

Reply:
2nd thought: I suggest to think about separate endpoints (rest and transport), the 2 "modes" are completely different and force you to program with modes, introducing `if`'s here and there. This is not straight forward and is more error-prone. I think separation is cleaner.

Reply:
update: I like the separation in _prepareRequest_.

Reply:
> Looking at data feeds I see both (POST, GET) specified for preview (user friendliness?). We should be consistent.

Done, I've added the missing `GET` paths.

Reply:
> 2nd thought: I suggest to think about separate endpoints (rest and transport), the 2 "modes" are completely different and force you to program with modes, introducing if's here and there. This is not straight forward and is more error-prone. I think separation is cleaner.

I think after I implemented changes suggested by Ben separating endpoints is not needed anymore.
WDYT?

Reply:
I agree, the separation in `prepareRequest` looks simple enough to me. I made the comment before you reverted the transport changes.